lib/web_ui/lib/src/engine/view_embedder/application_dom/application_dom.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../dom.dart';
+
+import '../hot_restart_cache_handler.dart';
+import 'custom_element_application_dom.dart';
+import 'full_page_application_dom.dart';
+
+/// Provides the API that the FlutterViewEmbedder uses to interact with the DOM.
+///
+/// The base class handles "global" stuff that is shared across implementations,
+/// like handling hot-restart cleanup.
+///
+/// This class is specialized to handle different types of DOM embeddings:
+///
+/// * [FullPageApplicationDom] - The default behavior, where flutter takes
+///   control of the whole web page. This is how Flutter Web used to operate.
+/// * [CustomElementApplicationDom] - Flutter is rendered inside a custom host
+///   element, provided by the web app programmer through the engine
+///   initialization.
+abstract class ApplicationDom {
+  @mustCallSuper
```

Comment:
This annotation is probably not needed in the constructor, as the default constructor will be called by default unless another one is called in the subclass?

Reply:
Yeah, I'm not sure this annotation has any effect on constructors.